zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java

```diff
@@ -670,6 +676,100 @@ public void shouldApplyOutputMappings() {
         .contains(tuple("adHocResult", "[1,2]", processInstanceKey));
   }
 
+  @Test
+  public void shouldSetAdHocSubProcessElementsVariable() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlClasspathResource("/processes/ad-hoc-sub-process-elements.bpmn")
```

Comment:
I opted to not create this process inline as it would get too complex to add different scenarios and task types. Also for https://github.com/camunda/ad-hoc-sub-process-phase-3/issues/13, we need to provide some input mappings with `fromAi()` function calls (potentially combined ones), so it's easier to maintain this as external file. The external file uses the same process/ad-hoc sub-process IDs though, so it does not demand further adaptions to helpers/constants.